hmy/api_backend.go

```diff
@@ -333,88 +334,63 @@ func (b *APIBackend) GetValidatorInformation(
-		},
-		CurrentVotingPower: []staking.VotePerShard{},
+		CurrentlyInCommittee: inCommittee,
+		Wrapper:              *wrapper,
+		Performance:          nil,
+		ComputedMetrics:      nil,
+		TotalDelegated:       wrapper.TotalDelegation(),
+		EPoSStatus: effective.ValidatorStatus(
+			inCommittee, wrapper.Status == effective.Active,
+		).String(),
 	}
+
+	snapshot, err := b.hmy.BlockChain().ReadValidatorSnapshotAtEpoch(
+		now, addr,
+	)
+
 	if err != nil {
 		return defaultReply, nil
 	}
 
-	signed, toSign, quotient, err := availability.ComputeCurrentSigning(snapshot, wrapper)
+	computed, err := availability.ComputeCurrentSigning(
+		snapshot, wrapper, shard.Schedule.BlocksPerEpoch(),
+	)
+
 	if err != nil {
 		return defaultReply, nil
 	}
+
 	stats, err := b.hmy.BlockChain().ReadValidatorStats(addr)
 	if err != nil {
 		return defaultReply, nil
 	}
 
-	defaultReply.CurrentSigningPercentage = staking.Computed{signed, toSign, quotient}
-	defaultReply.CurrentVotingPower = stats.VotingPowerPerShard
+	if defaultReply.CurrentlyInCommittee {
+		defaultReply.Performance = &staking.CurrentEpochPerformance{
+			CurrentSigningPercentage: *computed,
+		}
+		defaultReply.ComputedMetrics = stats
+	}
+
 	return defaultReply, nil
 }
 
 // GetMedianRawStakeSnapshot ..
-func (b *APIBackend) GetMedianRawStakeSnapshot() (*big.Int, error) {
-	candidates := b.hmy.BlockChain().ValidatorCandidates()
-	essentials := map[common.Address]effective.SlotOrder{}
-	blsKeys := map[shard.BlsPublicKey]struct{}{}
-	for i := range candidates {
-		validator, err := b.hmy.BlockChain().ReadValidatorInformation(
-			candidates[i],
-		)
-		if err != nil {
-			return nil, err
-		}
-		if validator.EPOSStatus != effective.Active {
-			continue
-		}
-		if err := validator.SanityCheck(
-			staking.DoNotEnforceMaxBLS,
-		); err != nil {
-			continue
-		}
-
-		validatorStake := big.NewInt(0)
-		for i := range validator.Delegations {
-			validatorStake.Add(
-				validatorStake, validator.Delegations[i].Amount,
-			)
-		}
-
-		found := false
-		for _, key := range validator.SlotPubKeys {
-			if _, ok := blsKeys[key]; ok {
-				found = true
-			} else {
-				blsKeys[key] = struct{}{}
-			}
-		}
-
-		if found {
-			continue
-		}
-
-		essentials[validator.Address] = effective.SlotOrder{
-			validatorStake,
-			validator.SlotPubKeys,
-		}
+func (b *APIBackend) GetMedianRawStakeSnapshot() (
+	*committee.CompletedEPoSRound, error,
+) {
+	round, err := committee.NewEPoSRound(b.hmy.BlockChain())
+	if err != nil {
+		return nil, err
 	}
-
-	instance := shard.Schedule.InstanceForEpoch(b.CurrentBlock().Epoch())
-	stakedSlots :=
-		(instance.NumNodesPerShard() - instance.NumHarmonyOperatedNodesPerShard()) *
-			int(instance.NumShards())
-	median, _ := effective.Compute(essentials, stakedSlots)
-	return median.TruncateInt(), nil
+	// NOTE We do this because for RPC presentation, we index from 1 instead of 0
+	// like how committee assignment does.
+	round.MaximumExternalSlotAllowed++
```

Comment:
why? this is just a count right? Like 320.

Reply:
Because like note mentions, otherwise it comes out off by 1 . Aka 39 open slots instead of 40. So the 320 would look like 319 without the ++

Reply:
then the func which returns it should do the job. Because "MaximumExternalSlotAllowed" naming meant a count, not an index. The user of it expect a count.